addons/EmpyPopTracker/EmpyPopTracker.lua

```diff
@@ -167,16 +167,22 @@ end
 
 EmpyPopTracker.generate_info = function(nm, key_items, items)
     local info = {
-        has_all_kis = true,
+        has_all_pops = true,
         text = ''
     }
 
     if nm.pops then
-        for _, key_item_data in pairs(nm.pops) do
-            local has_pop_ki = owns_key_item(key_item_data.id, key_items)
+        for _, pop_item_data in pairs(nm.pops) do
+            local has_pop = false
 
-            if not has_pop_ki then
-                info.has_all_kis = false
+            if pop_item_data.type == 'key item' then
+                has_pop = owns_key_item(pop_item_data.id, key_items)
+            elseif pop_item_data.type == 'item' then
+                has_pop = owns_item(pop_item_data.id, items)
+            end
+
+            if not has_pop then
+                info.has_all_pops = false
```

Comment:
The entire function body can be replaced with:
```lua
return {
    has_all_pops = not nm.pops or nm.pops:all(function(item)
        return pop.type == 'item' and owns_item(pop.id, items) or owns_key_item(pop.id, key_items)
    end)
    text = generate_text(nm, key_items, items, 1)
}
```

If it complains about `all` being `nil` then you need to do `T(nm.pops):all`.

Adjust as you see fit for readability.

Reply:
Nice, I like it and I have just pushed the change. Thanks for the speedy review!